src/audio/drc/drc_math_hifi3.c

```diff
@@ -62,35 +50,49 @@ static inline ae_f32 log10_fixed(ae_f32 x)
 	 * fpminimax(log10(x), 5, [|SG...|], [1/2;sqrt(2)/2], absolute);
 	 * max err ~= 6.088e-8
 	 */
-	int32_t e;
-	int32_t lshift;
 	ae_f32 exp; /* Q7.25 */
 	ae_f32 acc; /* Q6.26 */
 	ae_f32 tmp; /* Q6.26 */
+	ae_f64 tmp64;
+	int bit = 31 - AE_NSAZ32_L(x);
+	int32_t e = bit - 26;
 
-	x = rexp_fixed(x, 26, &e); /* Q2.30 */
+	x = AE_SRAA32(x, bit - 30);
 	exp = e << 25; /* Q_CONVERT_FLOAT(e, 25) */
 
 	if ((int32_t)x > (int32_t)ONE_OVER_SQRT2_Q30) {
-		lshift = drc_get_lshift(30, 30, 30);
-		x = drc_mult_lshift(x, ONE_OVER_SQRT2_Q30, lshift);
+		tmp64 = AE_MULF32R_LL(x, ONE_OVER_SQRT2_Q30);
+		/* drc_get_lshift(30, 30, 30) = 1 */
+		tmp64 = AE_SLAI64S(tmp64, 1);
```

Comment:
You could use #define macro for the magic shift values to know the are for a Qx to Qy conversion.

Reply:
if use macro then I can not use AE_SLAI64S, and if use AE_SLAA64S, the log10_fixed reduce cycles from 12.8% down to 12.2%, so better get back the lshift calculation for better code readability.